src/HttpHeader.cc

```diff
@@ -469,15 +469,12 @@ HttpHeader::parse(const char *header_start, size_t hdrLen, Http::ContentLengthIn
             break;      /* terminating blank line */
         }
 
-        HttpHeaderEntry *e;
-        if ((e = HttpHeaderEntry::parse(field_start, field_end)) == NULL) {
+        auto *e = HttpHeaderEntry::parse(field_start, field_end, owner);
```

Comment:
If we do not reassign e, then please use `const`:

```suggestion
        const auto e = HttpHeaderEntry::parse(field_start, field_end, owner);
```

Reply:
This begins a cascade of side effects via push_back inside addEntry() needing the type of HttpHeaderEntry::entries vector to match the const'ness of this auto. Fixing the const'ness is going to have to be left until later.